src/components/Coaching/CoachingDetail/MonthlyCommitment/useMonthlyCommitmentAverage.ts

```diff
@@ -0,0 +1,97 @@
+import { useMemo } from 'react';
+import { DateTime } from 'luxon';
+import { AccountList, Maybe } from 'src/graphql/types.generated';
+import { useMonthlyCommitmentSingleMonthQuery } from './MonthlyCommitment.generated';
+
+type MpdInfo = Pick<AccountList, 'activeMpdStartAt' | 'activeMpdFinishAt'>;
+
+interface MonthlyCommitmentAverageResult {
+  loading: boolean;
+  average: number | null;
+}
+
+/*
+ * Calculate the monthly commitment average for an account list. It represents the amount of net
+ * monthly commitments that the account list has gained per month. For example, if the account list
+ * went from $5000/month committed to $6000/month in 4 months of active MPD, the monthly commitment
+ * average would be $250/month.
+ */
+export const useMonthlyCommitmentAverage = (
+  accountListId: string,
+  mpdInfo: Maybe<MpdInfo>,
+): MonthlyCommitmentAverageResult => {
+  // Calculate the first month that the user was on active MPD
+  const startMonth = useMemo(() => {
+    if (!mpdInfo?.activeMpdStartAt) {
+      return null;
+    }
+
+    const startAt = DateTime.fromISO(mpdInfo.activeMpdStartAt);
+    if (startAt > DateTime.now()) {
+      // If the start at date is in the future, don't load commitment data for the start at date
+      return null;
+    }
+    return startAt.endOf('month');
```

Comment:
What if the end of the month is in the future?

Reply:
If the end month is in the future, it gets clamped to the end of the current month.